src/WrappedFd.hpp

```diff
@@ -0,0 +1,31 @@
+#pragma once
+
+#include <cstdio>
+#include <string>
+#include <utility>
+
+class WrappedFd {
+    FILE *fd_;
+
+    void close() {
+        if (fd_)
+            fclose(fd_);
+        fd_ = nullptr;
+    }
+
+public:
+    explicit WrappedFd(FILE *fd) : fd_(fd) {}
+    WrappedFd(const WrappedFd &) = delete;
+    WrappedFd &operator=(const WrappedFd &) = delete;
+    WrappedFd(WrappedFd &&other) noexcept : fd_(std::exchange(other.fd_, nullptr)) {}
+    WrappedFd &operator=(WrappedFd &&other) noexcept {
```

Comment:
It makes me wonder if we'd ever actually do assignment of these :)

Reply:
It's easy to do by mistake, if you accidentally wrote something that took a WrappedFd by value:

```void print_debug_to(WrappedFd fd);```

you'd double-close the FD... (the parameter itself is a copy of the passed-in value...ugh)

Also in general there's the "rule of 5" - if you define ANY of the special functions, define all of them. The table of which defaults get applied is COMPLICATED so best to spell out what you want.